aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/CredentialStoreActions.kt

```diff
@@ -32,19 +33,34 @@ object CredentialStoreActions : StoreActions {
                     credentialStore.saveCredential(credentials)
                     legacyCredentialStore.deleteCredential()
                 }
-                CredentialStoreEvent(CredentialStoreEvent.EventType.LoadCredentialStore())
+                CredentialStoreEvent(
+                    CredentialStoreEvent.EventType.LoadCredentialStore(
+                        AmplifyCredentialType.DEVICE_METADATA,
+                        null
+                    )
+                )
             } catch (error: CredentialStoreError) {
                 CredentialStoreEvent(CredentialStoreEvent.EventType.ThrowError(error))
             }
             logger.verbose("$id Sending event ${evt.type}")
             dispatcher.send(evt)
         }
 
-    override fun clearCredentialStoreAction() =
+    override fun clearCredentialStoreAction(amplifyCredentialType: AmplifyCredentialType, username: String?) =
         Action<CredentialStoreEnvironment>("ClearCredentialStore") { id, dispatcher ->
             logger.verbose("$id Starting execution")
             val evt = try {
-                credentialStore.deleteCredential()
+                when (amplifyCredentialType) {
+                    AmplifyCredentialType.AMPLIFY_CREDENTIAL -> {
+                        credentialStore.deleteCredential()
+                    }
+                    AmplifyCredentialType.DEVICE_METADATA -> {
+                        username?.let {
+                            credentialStore.deleteDeviceKeyCredential(it)
+                        }
+                    }
+                    AmplifyCredentialType.ASF_DEVICE_ID -> TODO()
```

Comment:
TODO() throws `NotImplementedError`
```suggestion
                    AmplifyCredentialType.ASF_DEVICE_ID -> {
                      // TODO ...
                    }
```